lib/Db/UserDAO.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace OCA\Password_Policy\Db;
+
+use OCP\IDBConnection;
+
+class UserDAO {
+
+    private $db;
+
+    public function __construct(IDBConnection $db) {
+        $this->db = $db;
+    }
+
+    public function findAllUsers() {
+        $sql = 'SELECT * FROM oc_users LEFT JOIN oc_password_policy_expiration ON oc_users.uid = oc_password_policy_expiration.uid';
+        $stmt = $this->db->prepare($sql);
+        $stmt->bindParam(1, $id, \PDO::PARAM_INT);
+        $stmt->execute();
+
+        $rows = $stmt->fetchAll();
+
+        $stmt->closeCursor();
+        return $rows;
+    }
+
+
+    /**
+     * Updates or inserts (if not exists) the expiration Data for a user.
+     * User-Name (uid) has to be set within the $fields array
+     *
+     * @param array $fields
+     *
+     * @return int number of affected rows
+     */
+    public function updateUserExpirationData(array $fields) {
+        $sql = "INSERT INTO oc_password_policy_expiration";
+        $sql .= " (" . implode(',', array_keys($fields)) .  ")";
+        $sql .= ' VALUES ("' . implode('","', $fields) .'")';
+        $sql .= " ON DUPLICATE KEY UPDATE ";
```

Comment:
This does not work on all supported databases.

Either use `$this->db->insertIfNotExist()` and depending on the result do the update or manually select before the update/insert